### DISCUSSION TITLE
What is best practice of managing Server Roles Positions?

### DISCUSSION THREAD

**Discussion by myroslav** [Lucky3028/terraform-provider-discord#125] [Q&A] [answered]
Hi,

I am trying to manage multiple Server Roles that I have in a server with `discord_role` resource, however, the `position` argument albeit optional, has `1` as the default value. This makes all roles I created attempt to "jump" to position 1 (very bottom) on every `apply` operation.

What would be the best way to manage position of the roles? I'd need them to keep their positions somehow without the need to force "in-place update" every time there is `apply` operation.

With best regards,

Myroslav

**Comment by Lucky3028** [comment]
Hello.

Sorry for the delay in replying to your post.

The behavior is currently in the specification, and I would appreciate it if you could raise an Issue.

**Comment by myroslav** [comment, marked as answer]
I'd managed to convince Terraform to start ignoring changes of role position with following snippet inside `discord_role` resource.

```terraform
resource discord_role this {

     ...

    lifecycle {
      ignore_changes = [
        # Ignore changes to position, e.g. because as soon as object is added,
        # others shipt (change their position)
        position,
      ]
    }
}
```
Terraform does see the change of attribute but only reads it out of Discord and stores the actual attribute value in state with each `terraform apply` while reporting following:

```log
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply" 
which may have affected this plan:

  # discord_role.this has changed
  ~ resource "discord_role" "this" {
        id          = "1234567890123456789"
        name        = "Role Name"
      ~ position    = 63 -> 64
        # (6 unchanged attributes hidden)
    }


Unless you have made equivalent changes to your configuration, or ignored the relevant attributes 
using ignore_changes, the following plan may include actions to undo or respond to these changes.
```

**Reply by Lucky3028** [reply]
That's wonderful solution! I don't know the param, lifecycle and ignore_changes.
